cpp/src/arrow/type.h

```diff
@@ -475,6 +475,15 @@ class ARROW_EXPORT StructType : public NestedType {
   Status Accept(TypeVisitor* visitor) const override;
   std::string ToString() const override;
   std::string name() const override { return "struct"; }
+
+  /// Returns null if name not found
+  std::shared_ptr<Field> GetChildByName(const std::string& name) const;
+
+  /// Returns -1 if name not found
+  int64_t GetChildIndex(const std::string& name) const;
```

Comment:
Child indices are `int` everywhere else, not `int64_t`, so this function should return `int` as well.